Lib/test/test_clinic.py

```diff
@@ -2895,6 +2909,93 @@ def test_cloned_func_with_converter_exception_message(self):
                 func = getattr(ac_tester, name)
                 self.assertEqual(func(), name)
 
+    def test_depr_star_new(self):
+        regex = re.escape(
+            "Passing positional arguments to _testclinic.DeprStarNew() is "
+            "deprecated. Parameter 'a' will become a keyword-only parameter "
+            "in Python 3.14."
+        )
+        with self.assertWarnsRegex(DeprecationWarning, regex):
+            ac_tester.DeprStarNew(None)
```

Comment:
```suggestion
        with self.assertWarnsRegex(DeprecationWarning, regex) as cm:
            ac_tester.DeprStarNew(None)
        self.assertEqual(cm.filename, __file__)
```

No need to test the lineno here.